src/core/storage/fileio/s3_api.hpp

```diff
@@ -286,11 +286,14 @@ std::ostream& reportS3Error(std::ostream& ss, const s3url& parsed_url,
                             S3Operation::ops_enum operation,
                             const Aws::Client::ClientConfiguration& config,
                             const Response& outcome) {
+  auto error = outcome.GetError();
   ss << "('" << parsed_url << ", proxy: '" << config.proxyHost << "', region: '"
      << config.region << "')"
      << " Error while performing " << S3Operation::toString(operation)
-     << ". Error Name: " << outcome.GetError().GetExceptionName()
-     << ". Error Message: " << outcome.GetError().GetMessage();
+     << ". Error Name: " << error.GetExceptionName()
```

Comment:
Aws didn't provide a good enum to string method to better render the error message. Often the time, exception name and message is empty. So we need to add HTTP code to better tace error on the client-side.